src/EditorFeatures/Next/Options/EditorConfigDocumentOptionsProvider.DocumentOptions.cs

```diff
@@ -10,37 +12,42 @@ internal sealed partial class EditorConfigDocumentOptionsProvider : IDocumentOpt
         private class DocumentOptions : IDocumentOptions
         {
             private ICodingConventionsSnapshot _codingConventionSnapshot;
+            private readonly IErrorLoggerService _errorLogger;
 
-            public DocumentOptions(ICodingConventionsSnapshot codingConventionSnapshot)
+            public DocumentOptions(ICodingConventionsSnapshot codingConventionSnapshot, IErrorLoggerService errorLogger)
             {
                 _codingConventionSnapshot = codingConventionSnapshot;
+                _errorLogger = errorLogger;
             }
 
             public bool TryGetDocumentOption(Document document, OptionKey option, out object value)
             {
                 var editorConfigPersistence = option.Option.StorageLocations.OfType<EditorConfigStorageLocation>().SingleOrDefault();
+                var namingEditorConfigPersistence = option.Option.StorageLocations.OfType<NamingEditorConfigStorageLocation>().SingleOrDefault();
 
-                if (editorConfigPersistence == null)
+                if (editorConfigPersistence == null && namingEditorConfigPersistence == null)
                 {
                     value = null;
                     return false;
                 }
 
-                if (_codingConventionSnapshot.TryGetConventionValue(editorConfigPersistence.KeyName, out value))
+                var allRawConventions = _codingConventionSnapshot.AllRawConventions;
+                try
                 {
-                    try
+                    if (namingEditorConfigPersistence != null)
                     {
-                        value = editorConfigPersistence.ParseValue(value.ToString(), option.Option.Type);
+                        value = EditorConfigNamingStyleParser.GetNamingStylesFromDictionary(allRawConventions);
                         return true;
                     }
-                    catch (Exception)
+                    else
                     {
-                        // TODO: report this somewhere?
-                        return false;
+                        return editorConfigPersistence.TryParseReadonlyDictionary(allRawConventions, option.Option.Type, out value);
```

Comment:
Isn't the point of such a refactoring like this that instead of this having naming-styles special case knowledge, that you'd delete NamingEditorConfigStorageLocation and make it be just a new EditorConfigStorageLocation(EditorConfigNamingStyleParser.GetNamingStylesFromDictionary)?